apps/datahub/src/app/search/search-filters/search-filters.component.html

```diff
@@ -1,8 +1,62 @@
-<div class="flex flex-wrap xl:flex-nowrap gap-2">
-  @for(filter of searchConfig; track filter.fieldName) {
-  <mel-datahub-filter-dropdown
-    [fieldName]="filter.fieldName"
-    [title]="filter.title | translate"
-  ></mel-datahub-filter-dropdown>
-  }
+<div class="flex flex-row gap-5">
+  <div class="flex flex-row">
+    @if(displayCount < searchConfig.length) {
+    <div class="flex gap-2 flex-wrap lg:flex-nowrap">
+      @for(filter of searchConfig | slice : 0 : 3; track filter.fieldName) {
+      <mel-datahub-filter-dropdown
+        [fieldName]="filter.fieldName"
+        [title]="filter.title | translate"
+      ></mel-datahub-filter-dropdown>
+      }
+    </div>
+    } @else {
+    <div class="flex flex-wrap gap-2 w-[640px]">
+      @for(filter of searchConfig; track filter.fieldName) {
+      <mel-datahub-filter-dropdown
+        [fieldName]="filter.fieldName"
+        [title]="filter.title | translate"
+      ></mel-datahub-filter-dropdown>
+      }
+    </div>
+    }
+  </div>
+  <div class="flex flex-row items-start gap-4 -ml-[2px] mt-2">
+    @if(displayCount !== searchConfig.length) {
+    <button
+      data-cy="filterExpandBtn"
+      type="button"
+      class="flex flex-row mel-simple-button"
+      (click)="showAll()"
+      translate
+    >
+      mel.datahub.search.filters.more
+      <mat-icon
+        class="material-symbols-outlined ml-1 align-middle leading-none mt-0.5"
+        >keyboard_arrow_down</mat-icon
+      >
+    </button>
+    } @else {
+    <button
+      data-cy="filterCloseBtn"
+      type="button"
+      class="flex flex-row mel-simple-button"
+      (click)="showAll()"
+      translate
+    >
+      mel.datahub.search.filters.reduce
+      <mat-icon
+        class="material-symbols-outlined ml-1 align-middle leading-none mt-0.5"
+        >keyboard_arrow_up</mat-icon
+      >
+    </button>
+    }
+    <button
```

Comment:
I tried to do this, it bothers me as well :disappointed: But after 2h on it I decided it was maybe not worth the time. Would you have a suggestion? I think I'm out of ideas. The issue is that the container of the filters grows when it's expanded, even if I put a max width, and it makes the two buttons move :/

Reply:
Hum, I see this could be achieved with a fixed width adding `justify-between w-56`. While this is not ideal, a better solution might need more modifications on the DOM, not sure. So maybe it's a quick win for now as it would also help the responsiveness.